### DISCUSSION TITLE
Show & tell: remark-prosemirror

### DISCUSSION THREAD

**Discussion by smoores-dev** [unifiedjs/unified#255] [Show and tell]
[@handlewithcare/remark-prosemirror](https://www.npmjs.com/package/@handlewithcare/remark-prosemirror)

I just published the first release of remark-prosemirror! It contains a remark plugin to convert markdown into a [ProseMirror](https://prosemirror.net) document, and an mdast utility to convert a Prosemirror document into mdast.

Working with unified/syntax-tree made this so manageable. The big challenge was converting between ProseMirror marks (how ProseMirror models inline formatting, like bold and links) to mdast's nested phrasing content nodes. I ended up using the same algorithm the ProseMirror uses to map marks to HTML elements in the browser, which works well enough. It sometimes produces edge cases where adjacent formatted text nodes don't have their formatting combined, which results in a technically correct output that can be a little challenging for a human to read. :shrug: The cool thing about unified is that if someone really wanted to, they could add an mdast transformer that looks for that edge case and resolves it by modifying the tree!

**Comment by ChristianMurphy** [comment]
Thanks for sharing @smoores-dev!
If you are interested, you could open a PR to add the plugin to the list https://github.com/remarkjs/remark/blob/main/doc/plugins.md

**Reply by smoores-dev** [reply]
Just did so, thank you!
